tensorflow/contrib/rnn/BUILD

```diff
@@ -8,12 +8,51 @@ exports_files(["LICENSE"])
 
 package(default_visibility = ["//tensorflow:__subpackages__"])
 
+load("//tensorflow:tensorflow.bzl", "tf_custom_kernel_library")
+load("//tensorflow:tensorflow.bzl", "tf_custom_op_library")
+load("//tensorflow:tensorflow.bzl", "tf_gen_op_libs")
+load("//tensorflow:tensorflow.bzl", "tf_gen_op_wrapper_py")
 load("//tensorflow:tensorflow.bzl", "cuda_py_tests")
 
+tf_custom_kernel_library(
+    name = "lstm_kernels",
+    srcs = [
+        "kernels/lstm_ops.cc"
+    ],
+    hdrs = [
+        "kernels/lstm_ops.h"
+    ],
+    gpu_srcs = [
+        "kernels/lstm_ops_gpu.cu.cc"
+    ],
+    deps = [
+        "//tensorflow/core:framework_headers_lib",
+        "//third_party/eigen3",
+    ],
+)
+
+tf_custom_op_library(
+    name = "python/ops/_lstm_ops.so",
+    srcs = [
+        "ops/lstm_ops.cc",
```

Comment:
hi @keveman

sorry dont have my laptop w me to try. but are u suggesting to put the .h file under the srcs field? i can give that a try, i was under impression that srcs were cc only and h files go under hdrs. thx.

> On Apr 28, 2016, at 3:55 PM, Manjunath Kudlur <notifications@github.com> wrote:
> 
> In tensorflow/contrib/rnn/BUILD:
> 
> > +    hdrs = [
> > +        "kernels/lstm_ops.h"
> > +    ],
> > +    gpu_srcs = [
> > +        "kernels/lstm_ops_gpu.cu.cc"
> > +    ],
> > +    deps = [
> > +        "//tensorflow/core:framework_headers_lib",
> > +        "//third_party/eigen3",
> > +    ],
> > +)
> > +
> > +tf_custom_op_library(
> > +    name = "python/ops/_lstm_ops.so",
> > +    srcs = [
> > +        "ops/lstm_ops.cc",
> I don't understand why you can't put ops/lstm_ops.cc, kernels/lstm_ops.cc and kernels/lstm_ops.h in the srcs here. Note that hdrs is needed only if you want your cc_library to export those headers to dependent libraries. Here, that is not the case.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>